### PR TITLE
Problem: solomachine and ibc-go are outdated

### PR DESCRIPTION
**Pull request by mmsqe**
👮🏻👮🏻👮🏻 !!!! REFERENCE THE PROBLEM YOUR ARE SOLVING IN THE PR TITLE AND DESCRIBE YOUR SOLUTION HERE !!!! DO NOT FORGET !!!! 👮🏻👮🏻👮🏻

# PR Checklist:

- [ ] Have you read the [CONTRIBUTING.md](https://github.com/crypto-org-chain/chain-main/blob/master/CONTRIBUTING.md)?
- [ ] Does your PR follow the [C4 patch requirements](https://rfc.zeromq.org/spec:42/C4/#23-patch-requirements)?
- [ ] Have you rebased your work on top of the latest master? 
- [ ] Have you checked your code compiles? (`make`)
- [ ] Have you included tests for any non-trivial functionality?
- [ ] Have you checked your code passes the unit tests? (`make test`)
- [ ] Have you checked your code formatting is correct? (`go fmt`)
- [ ] Have you checked your basic code style is fine? (`golangci-lint run`)
- [ ] If you added any dependencies, have you checked they do not contain any known vulnerabilities? (`go list -json -m all | nancy sleuth`)
- [ ] If your changes affect the client infrastructure, have you run the integration test?
- [ ] If your changes affect public APIs, does your PR follow the [C4 evolution of public contracts](https://rfc.zeromq.org/spec:42/C4/#26-evolution-of-public-contracts)?
- [ ] If your code changes public APIs, have you incremented the crate version numbers and documented your changes in the [CHANGELOG.md](https://github.com/crypto-org-chain/chain-main/blob/master/CHANGELOG.md)?
- [ ] If you are contributing for the first time, please read the agreement in [CONTRIBUTING.md](https://github.com/crypto-org-chain/chain-main/blob/master/CONTRIBUTING.md) now and add a comment to this pull request stating that your PR is in accordance with the [Developer's Certificate of Origin](https://github.com/crypto-org-chain/chain-main/blob/master/CONTRIBUTING.md#developer-certificate-of-originn).

Thank you for your code, it's appreciated! :)

### CODE REVIEW

File: integration_tests/install_solo_machine.nix [hunk 1]
```diff
@@ -1,21 +1,21 @@
 { stdenv, fetchurl, lib }:
 let
-  version = "v0.1.3";
+  version = "v0.1.4";
   srcUrl = {
     x86_64-linux = {
       url =
         "https://github.com/crypto-com/ibc-solo-machine/releases/download/${version}/ubuntu-latest-${version}.tar.gz";
-      sha256 = "sha256-Ui92r8J0I3jWGUvaI83rf9PMNyBW4yfnq98lcJ+1fkg=";
+      sha256 = "";
```
Comment (mmsqe) [inline integration_tests/install_solo_machine.nix:8]:
```suggestion
      sha256 = "sha256-wQix7ZeYTVD+yIWGr4F+XXnHbvXe2WZeNxG2lIDtmJ4=";
```

File: .github/workflows/audit.yml [hunk 2]
```diff
@@ -25,6 +25,7 @@ jobs:
     - uses: actions/checkout@v3
       with:
           submodules: true
+          persist-credentials: true
```
Comment (yihuang) [inline .github/workflows/audit.yml:28]: why do we need this?

Comment (mmsqe) [reply .github/workflows/audit.yml:28]: just follow the error [message](https://github.com/crypto-org-chain/chain-main/actions/runs/11357568574/job/31590653692), previous ci was failed

Comment (yihuang) [reply .github/workflows/audit.yml:28]: https://github.com/actions/checkout#usage
the doc suggests it defaults to true, maybe try to upgrade to v4?

**Review by yihuang — APPROVED** [review]

**Pull request merged**